compiler/cgen.nim

```diff
@@ -1099,7 +1099,7 @@ proc genProcBody(p: BProc; procBody: PNode) =
 proc isNoReturn(m: BModule; s: PSym): bool {.inline.} =
   sfNoReturn in s.flags and m.config.exc != excGoto
 
-proc genProcAux(m: BModule, prc: PSym) =
+proc genProcAux*(m: BModule, prc: PSym) =
```

Comment:
There is no need to export this, I think.

Reply:
It is used in pipelines because `createTypeBoundOps` needs `PContext` to work propely and the C code of method dispatcher cannot be generated in the cgen module. In pipelines, it needs to generate the code of the symbol. `genTopLevelStatement(m, disp.sym.ast)` is not enough. It has to use `genProcAux` like what is done before.

```nim
        let ctx = preparePContext(graph, module, idgen)
        for disp in disps:
          let retTyp = disp.sym.typ[0]
          if retTyp != nil:
            # todo properly semcheck the code of dispatcher?
            createTypeBoundOps(graph, ctx, retTyp, disp.info, idgen)
          genProcAux(BModule(bModule), disp.sym)
 ```